presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -419,111 +166,14 @@ public Block readBlock(Type type, int columnIndex)
         return block;
```

Comment:
The function `readBlock` contains pretty critical logic in terms of deciding the next batch size. Do we wanna do something similar to the selective reader? In general, I'm curious how selective reader can control memory consumption given the load size is quite unpredictable? Both Facebook and Twitter see severe reliability issues (e.g., humongous allocation) due to unpredictable load size. cc: @wenleix

Reply:
@highker Indeed, batch size adaptation is going to be non-trivial, hence, it is not included in this PR. At a high level, we'll be monitoring filter selectivity and average size of records that pass the filters and adjust the batch size accordingly.